### PR TITLE
Adding a set of preliminary rules for supervisor

### PR DESCRIPTION
**Pull request by kuqin12**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

This is the first iteration of rules that should pass all the image relocation and loading reversion for the supervisor.

Most "non zero" entries should be replaced by the "mem attr" entries in the near future.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested on Q35

## Integration Instructions

N/A

### CODE REVIEW

**Comment by kuqin12** [comment]
Hold off on this till https://github.com/microsoft/mu_feature_mm_supv/pull/241 merges.

**Comment by kuqin12** [comment]
> Hold off on this till #241 merges.

The prerequisite is merged, ready for this one now.

File: MmSupervisorPkg/Core/aux_config.spam.cfg [hunk 1]
```diff
@@ -1 +1,437 @@
...
+
+[[rule]]
+symbol = "mMapLevel"
+validation.type = "none"
+
+[[rule]]
+symbol = "mSmmCpuPrivateData"
+validation.type = "none"
+
+# [[rule]]
```
Comment (Javagedes) [inline MmSupervisorPkg/Core/aux_config.spam.cfg:279]: Should these be commented still?

File: MmSupervisorPkg/Core/aux_config.spam.cfg [hunk 2]
```diff
@@ -1 +1,437 @@
...
+
+[[rule]]
+symbol = "mXdSupportedAbsAddr"
+validation.type = "self"
+validation.reference = "mXdSupported"
+size = 8
+offset = -8
+
+[[rule]]
+symbol = "gcSmiIdtr"
```
Comment (Javagedes) [inline MmSupervisorPkg/Core/aux_config.spam.cfg:141]: Is there any functionality missing from the tool that is making you need to specify the size here?

File: MmSupervisorPkg/Core/aux_config.spam.cfg [hunk 3]
```diff
@@ -1 +1,437 @@
...
+validation.type = "non zero"
+
+[[rule]]
+symbol = "mFirstMmi"
+validation.type = "content"
+validation.content = [0x00]
+
+# TODO: Need to handle this one better
+[[rule]]
+symbol = "mMmCoreMmiHandlers"
```
Comment (Javagedes) [inline MmSupervisorPkg/Core/aux_config.spam.cfg:192]: Is there any functionality missing from the tool that is making you specify the size here?

**Review by Javagedes — APPROVED** [review]

File: MmSupervisorPkg/Core/aux_config.spam.cfg (hunk 3 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/aux_config.spam.cfg:232]: Good question, mMmCoreMmiHandlers is set as `MM_CORE_MMI_HANDLERS  mMmCoreMmiHandlers[]` with specified init value. We do not have a way to specify which entry in the list we want to validate, right? i.e. symbol = "mMmCoreMmiHandlers[2]", etc?

File: MmSupervisorPkg/Core/aux_config.spam.cfg (hunk 2 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/aux_config.spam.cfg:159]: This is an odd case when i initially tested it, where the size is 0. Let me revamp with the new version of tool.

File: MmSupervisorPkg/Core/aux_config.spam.cfg (hunk 1 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/aux_config.spam.cfg]: Nope, removed the comment

File: MmSupervisorPkg/Core/aux_config.spam.cfg (hunk 2 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/aux_config.spam.cfg:159]: Hmm, leaving the size to be automatically determined (Base or Limit) is causing the tool to fail somehow.

File: MmSupervisorPkg/Core/aux_config.spam.cfg (hunk 3 above)
Comment (Javagedes) [reply MmSupervisorPkg/Core/aux_config.spam.cfg:232]: Nope, the tool does not have that capability at this time!

Comment (kuqin12) [reply MmSupervisorPkg/Core/aux_config.spam.cfg:232]: No worries, we will get to this piece later in the process :)

File: MmSupervisorPkg/Core/aux_config.spam.cfg (hunk 2 above)
Comment (kuqin12) [reply MmSupervisorPkg/Core/aux_config.spam.cfg:159]: Fixed with the latest change from https://github.com/microsoft/mu_feature_mm_supv/pull/250

**Pull request merged**